src/fields/field_elementwise.cc

```diff
@@ -119,8 +119,11 @@ bool FieldElementwise<spacedim, Value>::set_time(const TimeStep &time) {
     //TODO: is it possible to check this before calling set_time?
     //if (time.end() == numeric_limits< double >::infinity()) return false;
     
+    BaseMeshReader::DiscretizationParams disc_params;
+    disc_params.discretization = BaseMeshReader::Discretization::element_data;
+    ReaderInstances::instance()->get_reader(reader_file_)->set_actual_data_header(field_name_, time.end(), disc_params);
```

Comment:
Rename 'DiscretizationParams' to 'HeaderQuery' which would be a structure including also the field name, and time.